pkg/monitor/intervalcreation/types.go

```diff
@@ -45,6 +50,18 @@ func InsertCalculatedIntervals(startingIntervals []monitorapi.EventInterval, rec
 	return ret
 }
 
+func CollectInClusterData(kubeConfig *rest.Config) (monitorapi.Intervals, error) {
```

Comment:
I believe there is already a mechanism for pickup intervals from out of process via files that start with an AdditionalEvents prefix, see the files in: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-upgrade/1663173231802060800/artifacts/e2e-aws-ovn-upgrade/openshift-e2e-test/artifacts/junit/

This is used for a few disruption backends that require complex setup/teardown, not unlike yours. If you could plug into that same framework, or at least name the files the same, they should automatically get picked up and this function is not needed.

Reply:
Ah, good catch, it really simplifies the code